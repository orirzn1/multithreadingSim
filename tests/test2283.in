L9
S8
O5
N5

T0
I@0x00000000
ADD $3, $1, $6
SUBI $7, $6, 0x1
STORE $3, $6, 0x2
STORE $0, $3, $1
SUBI $5, $1, 0x4
SUBI $1, $2, 0x4
ADDI $7, $5, 0x1
LOAD $4, $5, 0x3
HALT $0

T1
I@0x00000000
SUB $4, $6, $3
ADDI $1, $3, 0x3
STORE $6, $0, $6
LOAD $2, $6, 0x1
STORE $1, $0, $3
LOAD $0, $1, 0x4
HALT $0

T2
I@0x00000000
LOAD $2, $6, $0
SUBI $4, $7, 0x2
ADD $0, $6, $5
STORE $5, $7, 0x1
STORE $5, $2, $5
LOAD $3, $0, $0
HALT $0

T3
I@0x00000000
SUB $1, $3, $1
SUBI $4, $6, 0x3
SUBI $0, $7, 0x2
ADDI $6, $1, 0x4
STORE $1, $5, 0x4
HALT $0

T4
I@0x00000000
ADD $1, $3, $1
STORE $1, $5, 0x4
SUB $5, $1, $6
SUB $2, $0, $5
SUB $1, $4, $5
SUBI $7, $3, 0x4
LOAD $6, $0, $4
SUB $0, $7, $7
HALT $0

D@0x0
0x2
0x4
0x4
0x5
0x6
