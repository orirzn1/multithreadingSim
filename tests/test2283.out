
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0xFFFFFFFD
Register file thread id 1:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFE	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFC
Blocked MT CPI for this program 3.447368

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0xFFFFFFFD
Register file thread id 1:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFE	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFC
Finegrained Multithreading CPI for this program 1.289474

